geode-gfsh/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/RestoreRedundancyCommandDUnitTest.java

```diff
@@ -0,0 +1,618 @@
+
+    String command = new CommandStringBuilder(COMMAND_NAME).getCommandString();
+
+    CommandResultAssert commandResult = gfsh.executeAndAssertThat(command).statusIsSuccess()
+        .hasNoSection(ZERO_REDUNDANCY_SECTION)
+        .hasNoSection(UNDER_REDUNDANCY_SECTION);
+
+    InfoResultModelAssert summary = commandResult.hasInfoSection(SUMMARY_SECTION);
+    summary.hasOutput().contains(ZERO_REDUNDANT_COPIES + 0);
+    summary.hasOutput().contains(PARTIALLY_SATISFIED_REDUNDANCY + 0);
+    summary.hasOutput().contains(FULLY_SATISFIED_REDUNDANCY + regionNames.size());
+
+    InfoResultModelAssert satisfiedSection =
+        commandResult.hasInfoSection(SATISFIED_REDUNDANCY_SECTION);
+    satisfiedSection.hasHeader().isEqualTo(REDUNDANCY_SATISFIED_FOR_REGIONS);
+    satisfiedSection.hasOutput().contains(regionNames);
+
+    InfoResultModelAssert primariesSection = commandResult.hasInfoSection(PRIMARIES_INFO_SECTION);
+    primariesSection.hasOutput().contains(PRIMARY_TRANSFERS_COMPLETED, PRIMARY_TRANSFER_TIME);
+
+    // Confirm all regions have their configured redundancy and that primaries were balanced
+    int numberOfActiveServers = servers.size();
+    servers.get(0).invoke(() -> {
+      for (String regionName : regionNames) {
+        assertRedundancyStatusForRegion(regionName, true);
+        assertPrimariesBalanced(regionName, numberOfActiveServers, true);
+      }
+    });
+  }
+
+  @Test
+  public void restoreRedundancyWithIncludeRegionArgumentRestoresOnlyThoseRegions() {
+    List<String> regionNames = getAllRegionNames();
+    createAndPopulateRegions(regionNames);
+
+    int numberOfServers = servers.size();
+    regionNames.forEach(region -> locator
+        .waitUntilRegionIsReadyOnExactlyThisManyServers(SEPARATOR + region, numberOfServers));
+
+    String regionToInclude = HIGH_REDUNDANCY_REGION_NAME;
+    List<String> nonIncludedRegions = new ArrayList<>(regionNames);
+    nonIncludedRegions.remove(regionToInclude);
+
+    String command = new CommandStringBuilder(COMMAND_NAME)
+        .addOption(INCLUDE_REGION, regionToInclude).getCommandString();
+
+    CommandResultAssert commandResult = gfsh.executeAndAssertThat(command).statusIsSuccess()
+        .hasNoSection(ZERO_REDUNDANCY_SECTION)
+        .hasNoSection(UNDER_REDUNDANCY_SECTION)
+        .doesNotContainOutput(nonIncludedRegions.toArray(new String[0]));
+
+    InfoResultModelAssert summary = commandResult.hasInfoSection(SUMMARY_SECTION);
+    summary.hasOutput().contains(ZERO_REDUNDANT_COPIES + 0);
+    summary.hasOutput().contains(PARTIALLY_SATISFIED_REDUNDANCY + 0);
+    summary.hasOutput().contains(FULLY_SATISFIED_REDUNDANCY + 1);
+
+    InfoResultModelAssert satisfiedSection =
+        commandResult.hasInfoSection(SATISFIED_REDUNDANCY_SECTION);
+    satisfiedSection.hasHeader().isEqualTo(REDUNDANCY_SATISFIED_FOR_REGIONS);
+    satisfiedSection.hasOutput().contains(regionToInclude);
```

Comment:
This code block is duplicated within the test, it can be extracted to a separate method.